core/src/test/scala/org/apache/spark/storage/DiskBlockManagerSuite.scala

```diff
@@ -141,28 +142,48 @@ class DiskBlockManagerSuite extends SparkFunSuite with BeforeAndAfterEach with B
     assert(attemptId.equals("1"))
   }
 
+  // Use jnr to get and override the current process umask.
+  // Expects the input mask to be an octal number
+  private def getAndSetUmask(posix: POSIX, mask: String): String = {
+    val prev = posix.umask(BigInt(mask, 8).toInt)
```

Comment:
Is there any existing utility for setting umask? I thought Hadoop APIs had this somewhere and that we use it. No big deal if not. But if we have other places we use umask, could be good to standardize

Reply:
We dont currently get/set umask. I did see use of jna in hadoop, but that is shaded now.
Any library which allows us to set process umask would do actually, jnr simply seemed to be more easy to use across platforms (this is for tests anyway).

Reply:
The only place I know of is in the native code for the container executor